airflow/providers/amazon/aws/triggers/s3.py

```diff
@@ -0,0 +1,106 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import asyncio
+import re
+from functools import cached_property
+from typing import Any
+
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.amazon.aws.sensors.s3 import process_files
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+
+class S3KeyTrigger(BaseTrigger):
+    """Trigger for S3KeySensor"""
+
+    def __init__(
+        self,
+        *,
+        bucket_key: str | list[str],
+        bucket_name: str | None = None,
+        wildcard_match: bool = False,
+        aws_conn_id: str = "aws_default",
+        verify: str | bool | None = None,
+        poll_interval: int = 60,
+    ):
+        self.bucket_name = bucket_name
+        self.bucket_key = bucket_key
+        self.wildcard_match = wildcard_match
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.poll_interval = poll_interval
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        return (
+            "airflow.providers.amazon.aws.triggers.s3.S3KeyTrigger",
+            {
+                "bucket_key": self.bucket_key,
+                "bucket_name": self.bucket_name,
+                "wildcard_match": self.wildcard_match,
+                "aws_conn_id": self.aws_conn_id,
+                "verify": self.verify,
+                "poll_interval": self.poll_interval,
+            },
+        )
+
+    @cached_property
+    def hook(self) -> S3Hook:
+        return S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+
+    async def poke(self):
+        if isinstance(self.bucket_key, str):
+            self.bucket_keys = [self.bucket_key]
+        else:
+            self.bucket_keys = self.bucket_key
+
+        wildcard_keys = []
+        obj = []
+        bucket_key_names = []
+        for i in range(len(self.bucket_keys)):
```

Comment:
This loop looks very much like `poke` in `S3KeySensor`. Can they share one single implementation?